chapters/07_chapter_6.tex

```diff
@@ -0,0 +1,390 @@
+
+\subsection{Example: Cache attack on RSA Square-and-Multiply Exponentiation}
+
+\begin{algorithm}
+    \SetKwInOut{Input}{Input}
+    \SetKwInOut{Output}{Output}
+    \Input{base $b$, exponent $e$, modulus $n$}
+    \Output{$b^e$ mod $n$}
+    $X \leftarrow 1$
+    \For{$i \leftarrow bitlen(e)$ \textbf{downto} $0$}
+    {
+    $X \leftarrow multiply(X,X)$
+    \If{$e_i = 1$}
+    {
+    $X \leftarrow multiply(X,b)$
+    }
+    }
+    \textbf{return} X
+    \caption{Square-and-multiply exponentiation}
+    \label{algo:SaM}
+\end{algorithm}
+
+Consider the binary exponentiation algorithm presented in Algorithm \ref{algo:SaM}. Notice that the execution flow of the algorithm relies heavily on the value of the bit $e_i$ of the private key. Now consider a scenario in which an attacker has information on the changes regarding the buffer holding the multiplier $b$. Such information can be considered as a query to the buffer and receiving as a result the latency of the query. If the buffer is in use, the query will be longer then if the buffer is unused.
+
+\begin{figure}
+    \centering
+    \includegraphics[width=\textwidth]{images/PPSM.PNG}
+    \caption{Querying the buffer holding the multiplier $b$. The y axis is a latency scale, and the x axis represent the query index. The dotted plot is a scatter plot while the solid line is the normalized moving average.}
+    \label{fig:PPSQ}
+\end{figure}
+
+In Figure \ref{fig:PPSQ} the resulting graph of the querying can be seen, in which the actual bits of the exponent can be clearly detected.
+
+\subsection{Attack Model}
+We assume that the attacker has no physical access to the device. Moreover, the attacker can only execute unprivileged code on the same machine as victim. Such scenarios in which this happens can be found, for example, when the victim install some malicious program on his machine/smartphone. Another examples can be found in could computing where an attacker has a victual machine on the same physical machine as the victim, or in a web environment in which the attacker runs unprivileged JavaScript code.
+
+In the scope of this chapter we will focus on shared hardware in the form of data/instruction cache. While other shared hardware components can also include the DRAM and the memory bus (memory shared hardware) or the branch prediction unit and the arithmetic logic unit (CPU shared hardware), they are not in the scope of this chapter.
+
+\subsection{Today’s CPU Complexity}
```

Comment:
Please change to regular ' not this symbol, to avoid problems.

Reply:
Done